surface/include/pcl/surface/impl/mls.hpp

```diff
@@ -738,9 +738,20 @@ pcl::MLSResult::computeMLSSurface (const pcl::PointCloud<PointT> &cloud,
   model_coefficients.head<3> ().matrix () = eigen_vector;
   model_coefficients[3] = -1 * model_coefficients.dot (xyz_centroid);
 
+  query_point = cloud.points[index].getVector3fMap ().template cast<double> ();
+
+  if (!pcl_isfinite (eigen_vector[0]) || !pcl_isfinite (eigen_vector[1]) || !pcl_isfinite (eigen_vector[2]))
```

Comment:
We have a pending PR that removes `pcl_isfinite` macro (#2798). Please use `eigen_vector.isFinite()` instead.